src/DataFilterer.ts

```diff
@@ -329,4 +334,22 @@ export class DataFilterer {
         }, this);
         return summedMetricForDisagg;
     }
+
+    // This is a slightly hacky way to dynamically assign colours to keys that don't have them
+    // This should never be hit, if it is we should add the missing colours to ./PlotColours.ts
+    private getColour(key: string, colourDict: { [key: string]: string },
+                      niceColours:{ [key: string]: string }): void {
+        // check if this key is in the dictionary...
+        if (!(key in colourDict)) { //...if not try to find a new colour
+            console.log("Warning: " + key + " does not have a default colour");
+            // convert niceColours to an array
+            const extraCNames = Object.keys(niceColours);
+            // pick one at random
+            let colourName: string = extraCNames[Math.floor(Math.random()*extraCNames.length)];
```

Comment:
So I guess if you've run out of free nice colours, so its length is 0, this line blows up because you'll still be trying to access the value at position 0. Rather than leaving this (hopefully unlikely) danger in the code, would it be safer just to assign any old colour in this case (e.g. some mid grey) and log and error?